File: rust/src/models/endpoint_disabled_event_data.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::endpoint_disabled_trigger::EndpointDisabledTrigger;

/// Sent when an endpoint has been automatically disabled after continuous
/// failures, or manually via an API call.
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct EndpointDisabledEventData {
    /// The Application's ID.
    #[serde(rename = "appId")]
    pub app_id: String,

    /// The Application's UID.
    #[serde(rename = "appUid")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub app_uid: Option<String>,

    /// The Endpoint's ID.
    #[serde(rename = "endpointId")]
    pub endpoint_id: String,

    /// The Endpoint's UID.
    #[serde(rename = "endpointUid")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub endpoint_uid: Option<String>,

    #[serde(rename = "failSince")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub fail_since: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub trigger: Option<EndpointDisabledTrigger>,
}

impl EndpointDisabledEventData {
    pub fn new(app_id: String, endpoint_id: String) -> Self {
        Self {
            app_id,
            app_uid: None,
            endpoint_id,
            endpoint_uid: None,
            fail_since: None,
            trigger: None,
        }
    }
}

File: javascript/src/models/endpointDisabledEventData.ts
// this file is @generated
/* eslint @typescript-eslint/no-explicit-any: 0 */
import {
  EndpointDisabledTrigger,
  EndpointDisabledTriggerSerializer,
} from "./endpointDisabledTrigger";

/** Sent when an endpoint has been automatically disabled after continuous failures, or manually via an API call. */
export interface EndpointDisabledEventData {
  /** The Application's ID. */
  appId: string;
  /** The Application's UID. */
  appUid?: string | null;
  /** The Endpoint's ID. */
  endpointId: string;
  /** The Endpoint's UID. */
  endpointUid?: string | null;
  failSince?: Date | null;
  trigger?: EndpointDisabledTrigger;
}

export const EndpointDisabledEventDataSerializer = {
  _fromJsonObject(object: any): EndpointDisabledEventData {
    return {
      appId: object["appId"],
      appUid: object["appUid"],
      endpointId: object["endpointId"],
      endpointUid: object["endpointUid"],
      failSince: object["failSince"] ? new Date(object["failSince"]) : null,
      trigger: object["trigger"]
        ? EndpointDisabledTriggerSerializer._fromJsonObject(object["trigger"])
        : undefined,
    };
  },

  _toJsonObject(self: EndpointDisabledEventData): any {
    return {
      appId: self.appId,
      appUid: self.appUid,
      endpointId: self.endpointId,
      endpointUid: self.endpointUid,
      failSince: self.failSince,
      trigger: self.trigger
        ? EndpointDisabledTriggerSerializer._toJsonObject(self.trigger)
        : undefined,
    };
  },
};
